test/spec/cloudinary-spec.js

```diff
@@ -397,6 +397,54 @@ describe('Cloudinary', function() {
       });
     }
   });
+  describe("placeholder_url", function () {
+    const publicId = "sample";
+    let transformation;
+    let cl;
+    let config = {
+      'cloud_name': 'test123'
+    };
+
+    beforeEach(function () {
+      transformation = {};
+      cl = new cloudinary.Cloudinary(config);
+    });
+
+    it(`should generate blur placeholder image for undefined placeholder type`, function () {
+      const placeholder = cl.placeholder_url(publicId);
+      expect(placeholder).toEqual('http://res.cloudinary.com/test123/image/upload/e_blur:2000,f_auto,q_1/sample');
+    });
+    it(`should generate blur placeholder image for nonexistent placeholder type`, function () {
+      const placeholder = cl.placeholder_url(publicId, "custom");
+      expect(placeholder).toEqual('http://res.cloudinary.com/test123/image/upload/e_blur:2000,f_auto,q_1/sample');
+    });
+    it(`should generate blur placeholder image`, function () {
+      const placeholder = cl.placeholder_url(publicId, 'blur', transformation);
+      expect(placeholder).toEqual('http://res.cloudinary.com/test123/image/upload/e_blur:2000,f_auto,q_1/sample');
+    });
+    it(`should generate vectorize placeholder image`, function () {
+      const placeholder = cl.placeholder_url(publicId, 'vectorize', transformation);
```

Comment:
The transformation is empty, it doesn't test the merging logic

Reply:
i'll add a test that does